File: OCaml/lab/Lab1.ml
(*zad1*)
(*Napisac funkcje twoLast: 'a list -> 'a * 'a zwracajaca *)
(*przedostatni i ostatni element listy *)
let rec twoLast list =
	if list = [] then failwith "pusta"
	else if List.length list <2 then failwith "za mala"
	else if (List.length list = 2) then (List.hd list, List.hd(List.tl list))
	else
		twoLast(List.tl list)
;;

(*zad2*)
(*Napisz funkcje dwaRowne
  sprawdzajaca, czy na liscie sa dwa kolejne elementy rowne *)
let rec dwaRowne x = 
	if x = [] || List.tl x = [] then false
	else if List.hd x = List.hd(List.tl x) then true
	else
		dwaRowne(List.tl x)
;;

File: Scala/lab/Lab1.scala
//zad1
//Napisaæ funkcjê twoLast: ‘a list -> ‘a * ‘a zwracaj¹c¹
//przedostatni i ostatni element listy

def twoLast [A](list: List[A]) : (A,A)=
  if(list == Nil) throw new Exception ("pusta")
  else if (list.length <2) throw new Exception ("za mala")
  else if(list.length == 2)
     (list.head, (list.tail).head)
  else
     twoLast (list.tail)
     
//zad2
//Napisz funkcjê dwaRowne[A](xs:List[A]):Boolean
//sprawdzaj¹c¹, czy na liœcie s¹ dwa kolejne elementy równe     
def dwaRowne[A] (list: List[A]): Boolean = 
  if (list == Nil || list.tail == Nil) false
  else if (list.head == (list.tail).head) true